example-library/src/main/java/uk/gov/gchq/palisade/example/config/ExamplePolicyCacheWarmerFactory.java

```diff
@@ -0,0 +1,274 @@
+     * @param resource      a {@link String} value of the {@link Resource} to be used.
+     * @param owner         a {@link String} value of the owner of the policy
+     * @param resourceRules a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     * @param recordRules   a {@link Map} containing the ({@link String}) message and the ({@link String}) rule name.
+     */
+    public ExamplePolicyCacheWarmerFactory(final String type, final String resource, final String owner,
+                                           final Map<String, String> resourceRules, final Map<String, String> recordRules) {
+        this.type = type;
+        this.resource = resource;
+        this.owner = owner;
+        this.resourceRules = resourceRules;
+        this.recordRules = recordRules;
+    }
+
+    @Generated
+    public String getType() {
+        return type;
+    }
+
+    @Generated
+    public void setType(final String type) {
+        requireNonNull(type);
+        this.type = type;
+    }
+
+    @Generated
+    public String getResource() {
+        return resource;
+    }
+
+    @Generated
+    public void setResource(final String resource) {
+        requireNonNull(resource);
+        this.resource = resource;
+    }
+
+    @Generated
+    public String getOwner() {
+        return owner;
+    }
+
+    @Generated
+    public void setOwner(final String owner) {
+        requireNonNull(owner);
+        this.owner = owner;
+    }
+
+    @Generated
+    public Map<String, String> getResourceRules() {
+        return resourceRules;
+    }
+
+    @Generated
+    public void setResourceRules(final Map<String, String> resourceRules) {
+        requireNonNull(resourceRules);
+        this.resourceRules = resourceRules;
+    }
+
+    @Generated
+    public Map<String, String> getRecordRules() {
+        return recordRules;
+    }
+
+    @Generated
+    public void setRecordRules(final Map<String, String> recordRules) {
+        requireNonNull(recordRules);
+        this.recordRules = recordRules;
+    }
+
+    @Override
+    public Entry<Resource, Policy> policyWarm(final List<? extends UserCacheWarmerFactory> users) {
+        Policy<Employee> policy = new Policy<>();
+        for (ExampleUserCacheWarmerFactory user : (List<ExampleUserCacheWarmerFactory>) users) {
+            if (user.getUserId().equals(owner)) {
+                policy.owner(user.userWarm());
+            }
+        }
+        for (String key : resourceRules.keySet()) {
+            policy.resourceLevelRule(key, (Rule<Resource>) createRule(resourceRules.get(key), "resource"));
+        }
+        for (String key : recordRules.keySet()) {
+            policy.recordLevelRule(key, (Rule<Employee>) createRule(recordRules.get(key), "record"));
+        }
+        return new MapEntry<>(createResource(), policy);
+    }
+
+    private Rule<?> createRule(final String rule, final String ruleType) {
+        if ("resource".equalsIgnoreCase(ruleType)) {
```

Comment:
See similar services PR comment, either `Map<Exception, String> ERROR_MESSAGES` or general `catch (Exception ex) {LOGGER.error("There was an error: ", ex)}` with 'proper' passing of exception to log function